checker.cpp

```diff
@@ -2,21 +2,33 @@
 #include <iostream>
 using namespace std;
 
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    cout << "Temperature out of range!\n";
-    return false;
-  } else if(soc < 20 || soc > 80) {
-    cout << "State of Charge out of range!\n";
-    return false;
-  } else if(chargeRate > 0.8) {
-    cout << "Charge Rate out of range!\n";
-    return false;
-  }
-  return true;
+namespace
+{
+  const float tempr_min = 0.0F, tempr_max = 45.0F;
+  const float soc_min = 20.0F, soc_max = 80.0F;
+  const float charge_rate_max = 0.8F;
+}
+bool isValueOutOfRange (float value, float min, float max)
+{
+  bool valueOutOfRange = (value < min) || (value > max);
+  return valueOutOfRange;
+}
+
+bool batteryIsOk(float temperature, float soc, float chargeRate) 
+{
+  bool tempCheck = isValueOutOfRange(temperature , tempr_min, tempr_max);
+  bool socCheck = isValueOutOfRange(soc , soc_min, soc_max);
+  bool batteryIsNotOkay = tempCheck || socCheck || chargeRate > charge_rate_max;
```

Comment:
The complexity here increases with the number of parameters. Any way to avoid that? Think of how we describe the solution: "battery not ok if any of the parameters are not ok"